snapcraft/explanation/parts-lifecycle.md

```diff
@@ -0,0 +1,90 @@
+1. **pack**: takes the entire contents of the `prime` directory and packs it into [a snap](/t/the-snap-format/698).
+
+<h3 id='#heading--steps-commands'>Steps commands<sup><a href=##heading--steps-commands>⚓</a></sup></h3>
+
+Each of these lifecycle steps can be run from the command line, and the command can be part specific or apply to all parts in a project.
+
+1. `snapcraft pull [<part-name>]`
+1. `snapcraft build [<part-name>]`
+1. `snapcraft stage [<part-name>]`
+1. `snapcraft prime [<part-name>]`
+1. `snapcraft pack` or `snapcraft`
+
+Note that each command also executes the previous lifecycle steps, so `snapcraft` executes all the lifecycle steps chained together.
+
+To access the part environment at any stage, add the `--shell` argument. For example, `snapcraft prime --shell` will run up to the *prime* step and open a shell. See [Iterating over a build](/t/iterating-over-a-build/12143) for more details.
+
+<h3 id='heading--steps-dependencies'>Steps dependencies<sup><a href=#heading--steps-dependencies>⚓</a></sup></h3>
+
+Each lifecycle step depends on the completion of the previous step for that part, so to reach a desired step, all prior steps need to have successfully run. By default, `snapcraft` runs the same lifecycle step of all parts before moving to the next step. However, you can change this behavior using the `after` keyword in the definition of a part in `snapcraft.yaml`. This creates a dependency chain from one part to another.
+
+```yaml
+ grv:
+    plugin: go
+    go-channel: 1.11/stable
+    after:
+      - libgit2
+```
+
+In the above example, the part named `grv` will be built after the part named `libgit2` has been successfully built _and_ staged.
+
+<h3 id='heading--overriding-steps'>Overriding a step<sup><a href=#heading--overriding-steps>⚓</a></sup></h3>
+
+Each plugin defines the default actions that happen during a step. This behavior can be changed in two ways.
+
+- By using `override-<step-name>` in `snapcraft.yaml`. See [Overriding steps](/t/scriptlets/4892) for more details.
+- By using a local plugin.  This can inherit the parent plugin or scaffolding from the original. See [Local plugins](/t/writing-local-plugins/5125) for more details.
```

Comment:
It may be prudent to mention that this hasn't been supported for recent bases in snaps for several years -- I believe the cut-off is core20?